app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java

```diff
@@ -251,4 +246,14 @@ public void startScan() {
     public void stopScan() {
         cameraSource.stop();
     }
+
+    public void showToggleDialog(long attendeeId) {
+        BottomSheetDialogFragment bottomSheetDialogFragment = AttendeeCheckInFragment.newInstance(attendeeId);
+        bottomSheetDialogFragment.show(getSupportFragmentManager(), bottomSheetDialogFragment.getTag());
+    }
+
+    @Override
+    public void onDialogDismissed() {
+        presenter.allowScan(true);
```

Comment:
>Just change boolean to false when dialog is shown and to true when it is hidden

if we wait till dialog is shown to set boolean false, it let presenter to create more than one dialogs. Hence the boolean is set to false in filter itself and changed to true if attendee is not found, hence all that conditions. You may take a look again.